core/pkg/log/log.go

```diff
@@ -126,6 +126,17 @@ func Debugf(format string, a ...interface{}) {
 	log.Debug().Msgf(format, a...)
 }
 
+func DedupedDebugf(logTypeLimit int, format string, a ...interface{}) {
```

Comment:
I'd like to avoid increasing the use of `Deduped` log functions if it is at all possible. Swallowing dupes based on the format string makes these functions behave in unexpected ways. It is usually possible to achieve better behavior by just not logging in hot loops and instead accumulating error information outside of the loop to log at completion; this is superior deduping behavior.

Reply:
That's a valid point, I agree with your alternative solution.